scvi/lightning/_trainingplans.py

```diff
@@ -522,3 +523,55 @@ def validation_step(self, batch, batch_idx, optimizer_idx=0):
             "kl_global": scvi_losses.kl_global,
             "n_obs": reconstruction_loss.shape[0],
         }
+
+
+class PyroTrainingPlan(pl.LightningModule):
+    """
+    Lightning module task to train Pyro scvi-tools modules.
+
+    Parameters
+    ----------
+    pyro_module
+        A model instance from class ``PyroBaseModuleClass``.
+    lr
+        Learning rate used for optimization.
+    """
+
+    def __init__(
+        self,
+        pyro_module: PyroBaseModuleClass,
+        lr: float = 1e-3,
+        loss_fn: Callable = pyro.infer.Trace_ELBO(),
```

Comment:
Warning: it's dangerous to set stateful default arguments. In this case the same default `Trace_ELBO` instance will be shared by all models and (details below) bad things may happen. I'd recommend defaulting to `None` and doing a standard `if loss_fn is None: loss_fn = pyro.infer.Trace_ELBO()`.

<details>

Each `Trace_ELBO` instance guesses and stores the number of plates in its model, and assumes it will be associated with only a single model. If you use it with a different model with a different number of plates, you might see tensor shape errors.

</details>